File: backend/app/schemas/calendar.py
from __future__ import annotations

from datetime import date, time
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

CalendarEventType = Literal["training", "match", "task_deadline"]


class CalendarEventOut(BaseModel):
    id: UUID
    type: CalendarEventType
    date: date
    time: time | None
    title: str
    team_id: UUID | None
    team_name: str | None
    status: str

File: frontend/src/types/calendar.ts
import type { IconName } from "../components/shared/Icon";

export type CalendarEventType = "training" | "match" | "task_deadline";

export const CALENDAR_EVENT_ICONS: Record<CalendarEventType, IconName> = {
  training: "dumbbell",
  match: "ball",
  task_deadline: "clipboard",
};

export interface CalendarEvent {
  id: string;
  type: CalendarEventType;
  date: string;
  time: string | null;
  title: string;
  teamId: string | null;
  teamName: string | null;
  status: string;
}

export interface CalendarEventDto {
  id: string;
  type: CalendarEventType;
  date: string;
  time: string | null;
  title: string;
  team_id: string | null;
  team_name: string | null;
  status: string;
}

export function mapCalendarEventDto(dto: CalendarEventDto): CalendarEvent {
  return {
    id: dto.id,
    type: dto.type,
    date: dto.date,
    time: dto.time,
    title: dto.title,
    teamId: dto.team_id,
    teamName: dto.team_name,
    status: dto.status,
  };
}
